extensions/windows-event-log/Bookmark.cpp

```diff
@@ -50,56 +50,52 @@ Bookmark::Bookmark(const std::wstring& channel, const std::wstring& query, const
     state_manager_->set(state_map);
   }
 
-  if (!(hBookmark_ = EvtCreateBookmark(0))) {
+  if (!(hBookmark_ = unique_evt_handle{ EvtCreateBookmark(0) })) {
     LOG_LAST_ERROR(EvtCreateBookmark);
     return;
   }
 
-  const auto hEventResults = EvtQuery(0, channel.c_str(), query.c_str(), EvtQueryChannelPath);
+  const auto hEventResults = unique_evt_handle{ EvtQuery(0, channel.c_str(), query.c_str(), EvtQueryChannelPath) };
   if (!hEventResults) {
     LOG_LAST_ERROR(EvtQuery);
     return;
   }
-  const utils::ScopeGuard guard_hEventResults([hEventResults]() { EvtClose(hEventResults); });
 
-  if (!EvtSeek(hEventResults, 0, 0, 0, processOldEvents? EvtSeekRelativeToFirst : EvtSeekRelativeToLast)) {
+  if (!EvtSeek(hEventResults.get(), 0, 0, 0, processOldEvents? EvtSeekRelativeToFirst : EvtSeekRelativeToLast)) {
     LOG_LAST_ERROR(EvtSeek);
     return;
   }
 
-  DWORD dwReturned{};
-  EVT_HANDLE hEvent{};
-  if (!EvtNext(hEventResults, 1, &hEvent, INFINITE, 0, &dwReturned)) {
-    LOG_LAST_ERROR(EvtNext);
-    return;
-  }
+  const unique_evt_handle hEvent = [this,&hEventResults] {
+    DWORD dwReturned{};
```

Comment:
Why do we create a lambda for only getting something once?

Reply:
To be able to make it const and hide the complexity of out params and the returned array of length 1.

https://www.bfilipek.com/2016/11/iife-for-complex-initialization.html